### ISSUE TITLE
Default REPL output is difficult to read on light background

### ISSUE THREAD

**Issue by eschnett** [JuliaLang/julia#28690]
I'm using Julia on macOS, and my terminals have a light (almost white) background. The default color scheme of the REPL is difficult to read there, as e.g. yellow (for warnings) is almost unreadable.

[This page](https://unix.stackexchange.com/questions/245378/common-environment-variable-to-set-dark-or-light-terminal-background) has some hints for how to automatically detect whether the terminal background is light or dark.

**Comment by KristofferC** [comment]
For reference, the following env vars can be set to customize the colors, e.g.:

```
ENV["JULIA_ERROR_COLOR"] = :magenta
ENV["JULIA_WARN_COLOR"] = :yellow
ENV["JULIA_INFO_COLOR"] = :cyan
```

**Comment by eschnett** [comment]
Yes, I'm doing this. There are two problems: (1) It's not the default, and (2) this doesn't affect other parts of Julia, e.g. the package manager which still uses light grey for its status output.

I'm thinking that a function `has_light_background` in REPL's `Terminals.jl` (next to `hascolor` there) would make sense?

**Comment by ckoe-bccms** [comment]
Yes, having readable pkg status would be great. Eventually forcing everything to use  just the terminals foreground color (effectively monochrome) might be the simplest temporary solution to do ?

**Comment by KristofferC** [comment]
You could always start julia with `--color=no`.

**Comment by ckoe-bccms** [comment]
Ah, sorry for the noise then !

**Comment by vtjnash** [comment]
While the PR to autodetect doesn't seem like it will work well, so we haven't merged it in a couple years, I think the `--color=light` / `--color=dark` options presented there still have merit, if someone wanted to further develop those idea (there's still a couple days left in [hacktoberfest](https://hacktoberfest.digitalocean.com))

**Comment by newptcai** [comment]
I change the default to

~~~
ENV["JULIA_WARN_COLOR"] = :cyan
~~~

But it does not seem to have any effect.

**Comment by adigitoleo** [comment]
Adding some suggestions to the discussion here, after I went through the same experience:
- Present the `--color=no` option more visibly in the docs (e.g. at https://docs.julialang.org/en/v1/stdlib/REPL/)
- The `--color=light` or `--color=dark` would help, IIRC IPython has something similar
- A way to list all REPL element names (similar to `Base.text_colors` but showing the element->color mapping) would be useful (maybe I missed it?) for figuring out what to put in the startup config.

**Comment by adigitoleo** [comment]
For reference, here's a quick snippet to get all the "named" colors:

```julia
julia> for key in keys(Base.text_colors)
           if isa(key, Symbol)
               println(key)
           end
       end
```

The names match the base 16 unix/X11/terminal/(?) color names, so you can use a command like (Linux):

```
msgcat --color=test|head -11
```
and compare the colors to see which ones you want to change. Preview changes by reassigning to the color dict, e.g.:

```
julia> Base.text_colors[:white] = Base.text_colors[:red]
```

When you're happy put the ENV declarations in the config file (see prev comments). So this kind of solves the last bullet point in my other comment.

**Comment by tecosaur** [comment]
#49586 gave us some new tools to help with this, but I don't think we can consider it quite solved yet.
